ffi/src/lib.rs

```diff
@@ -1666,6 +1666,14 @@ pub unsafe extern "C" fn typed_value_value_type_kw(typed_value: *mut Binding) ->
     string_to_c_char(value.clone())
 }
 
+/// Returns the size of a `row` as usize
+#[no_mangle]
+pub unsafe extern "C" fn typed_value_list_size(values: *mut Vec<Binding>) -> c_int {
+    assert_not_null!(values);
+    let values = &*values;
+    values.len () as c_int
+}
+
 /// Returns the value at the provided `index` as a `Vec<ValueType>`.
```

Comment:
nit: no space between function and arguments, so: `.len()`.

Reply:
The formatting is done. I am happy to try and implement your suggestion but I don't understand. Please could you walk me through it? What should all the return types be? Should the java class size method return a String?

Reply:
Ok, my current idea would be to do this by passing a direct allocated static final byteBuffer from java. To do this it seems easiest if I have access to JNIEnv, is there a reason why you don't seem to use this style?
https://mozilla.github.io/firefox-browser-architecture/experiments/2017-09-21-rust-on-android.html

Would need to use synchronise(this) around the call to rust within the constructor.